Code/GraphMol/MolAlign/Wrap/rdMolAlign.cpp

```diff
@@ -72,12 +72,15 @@ PyObject *generateRmsdTransMatchPyTuple(double rmsd,
   PyTuple_SetItem(resTup, 0, rmsdItem);
   PyTuple_SetItem(resTup, 1, PyArray_Return(res));
   if (match) {
-    python::list pairList;
-    for (const auto &pair : *match) {
-      pairList.append(python::make_tuple(pair.first, pair.second));
+    PyObject *listTup = PyTuple_New(match->size());
+    for (unsigned int i = 0; i < match->size(); ++i) {
+      auto *pairTup = PyTuple_New(2);
+      PyTuple_SetItem(pairTup, 0, PyLong_FromLong((*match)[i].first));
+      PyTuple_SetItem(pairTup, 1, PyLong_FromLong((*match)[i].second));
+      PyTuple_SetItem(listTup, i, pairTup);
     }
```

Comment:
was there a leak here? This is a lot more verbose than the original formulation

Reply:
Yes there was. The `python::tuple` (or part of it) was leaking. I don't know exactly what was going on, but I'm guessing that despite we're attaching the internal `PyTuple` objet to `resTup`, there's some additional wrappings in the `python::tuple` object that were leaked.